packages/react-data-grid/src/common/editors/EditorContainer.tsx

```diff
@@ -1,274 +1,230 @@
-import React, { KeyboardEvent } from 'react';
+import React, { KeyboardEvent, useRef, useState, useEffect, useCallback, createElement, cloneElement } from 'react';
 import classNames from 'classnames';
 import { isElement, isValidElementType } from 'react-is';
 import { Clear } from '@material-ui/icons';
 
 import { CalculatedColumn, Editor, EditorProps, CommitEvent, Dimension, Omit } from '../types';
-import SimpleTextEditor from './SimpleTextEditor';
+import TextEditor from './TextEditor';
+import LegacyTextEditor from './LegacyTextEditor';
 import ClickOutside from './ClickOutside';
 import { InteractionMasksProps, InteractionMasksState } from '../../masks/InteractionMasks';
 
 type SharedInteractionMasksProps<R, K extends keyof R> = Pick<InteractionMasksProps<R, K>, 'scrollLeft' | 'scrollTop'>;
 type SharedInteractionMasksState = Pick<InteractionMasksState, 'firstEditorKeyPress'>;
 
-type ValueType<R> = R[keyof R];
-
-export interface Props<R, K extends keyof R> extends SharedInteractionMasksProps<R, K>, SharedInteractionMasksState, Omit<Dimension, 'zIndex'> {
+export interface EditorContainerProps<R, RK extends keyof R, CK extends keyof R> extends SharedInteractionMasksProps<R, RK>, SharedInteractionMasksState, Omit<Dimension, 'zIndex'> {
   rowIdx: number;
   rowData: R;
-  value: ValueType<R>;
-  column: CalculatedColumn<R>;
-  onGridKeyDown?(e: KeyboardEvent): void;
-  onCommit(e: CommitEvent<R>): void;
+  value: R[CK];
+  column: CalculatedColumn<R, unknown, CK>;
+  onCommit(e: CommitEvent<R, { [k: string]: R[CK] }>): void;
   onCommitCancel(): void;
 }
 
-interface State {
-  isInvalid: boolean;
-}
-
-export default class EditorContainer<R, K extends keyof R> extends React.Component<Props<R, K>, State> {
-  static displayName = 'EditorContainer';
+function getInitialValue<R, CK extends keyof R>({ key, value }: { key: string | null; value: R[CK] }): R[CK] | string {
+  // TODO: Do we need to activate editor when Delete, Backspace or other keys are pressed?
```

Comment:
I think in most cases pressing delete/backspace on a cell should wipe the value, update the row data, then open the editor.
There will be cases where a cell's value cannot be deleted, so I think we should add an optional `onDelete`/`onClear`/`onReset` handler, or something like that.

```js
const column = {
  ...column,
  onReset({ commit, ...data }) {
    commit(DEFAULT_VALUE);
    // commit(0);
    // commit('');
    // commit(null);
    return false; // optional, returning false would not open the editor
  }
};

// and/or...
const column = {
  ...,
  defaultValue: DEFAULT_VALUE // or 0 / '' / null / etc...
};
```

I guess the handler/default value would have to be mandatory for editable columns.